core/controllers/collection_editor.py

```diff
@@ -319,7 +319,8 @@ class ExplorationMetadataSearchHandler(base.BaseHandler):
     @acl_decorators.open_access
     def get(self):
         """Handles GET requests."""
-        query_string = base64.b64decode(self.normalized_request.get('q'))
+        q = self.normalized_request.get('q').encode('utf-8')
+        query_string = base64.b64decode(q).decode('utf-8')
```

Comment:
Can you explain conceptually how the encode/decode works here? It seems weird to me that there's a b64decode operation between them, but perhaps I haven't grokked yet how to handle Python 3 strings in our codebase.

(Is there a wiki page explaining that? It will probably trip up many devs.)

Reply:
The query (`q`) is encoded into base64 in the frontend. `b64decode` accepts and returns `bytes`, we thus firstly need to encode the base64 `str` query to `bytes` and after we decode the base64 we need to decode the returned `bytes` back to `str` so that we can work with it further.

Reply:
There is no wiki for this.

Reply:
Done elsewhere too.